mixins/test/label-mixin.test.js

```diff
@@ -0,0 +1,162 @@
+
+import { defineCE, expect, fixture, html, nextFrame } from '@open-wc/testing';
+import { LabelledMixin, LabelMixin } from '../labelled-mixin.js';
+import { LitElement } from 'lit-element/lit-element.js';
+
+const labelledTag = defineCE(
+	class extends LabelledMixin(LitElement) {
+		render() {
+			return html`
+				<input type="text" aria-label="${this.label}">
+			`;
+		}
+	}
+);
+
+const labelTag = defineCE(
+	class extends LabelMixin(LitElement) {
+		static get properties() {
+			return {
+				text: { type: String }
+			};
+		}
+		render() {
+			return html`
+				<span>${this.text}</span>
+			`;
+		}
+		updated(changedProperties) {
+			super.updated(changedProperties);
+			this.updateLabel(this.text);
+		}
+	}
+);
+
+describe.only('LabelMixin', () => {
+
+	let elem;
+
+	beforeEach(async() => {
+		elem = await fixture(`
+			<${labelTag} text="the label value"></${labelTag}>
+		`);
+	});
+
+	it('reflects label value', async() => {
+		expect(elem.getAttribute('_label')).to.equal('the label value');
+	});
+
+	it('reflects label value change', async() => {
+		elem.text = 'new label value';
+		await nextFrame();
```

Comment:
What's the difference between `nextFrame` and `await elem.updateComplete`? I think I usually do the latter.

Reply:
So at least for these tests, waiting for `updateComplete` is not sufficient for the browser to have applied the updates, which is necessary to assert the attribute values.  If these tests are changed to assert the property values (instead of the attribute values) then waiting for `updateComplete` is sufficient.  In some of these tests, I think checking the attribute value is preferred due to the way the mixins depend on attribute changes.  

I think this is due to the Lit timing for property changes and when those changes are reflected to the DOM, and the `nextFrame` helper uses rAF (task) so it's waiting longer than just `updateComplete` (microtask I think).

Reply:
Gotcha makes sense thx.